### PR TITLE
User cannot create/delete a collection owned by another user

### PR DESCRIPTION
**Pull request by agrosu1978**

### CODE REVIEW

File: services/catalog/rs_server_catalog/user_catalog.py [hunk 1]
```diff
@@ -448,6 +452,16 @@ async def manage_put_post_request(self, request: Request) -> Request | JSONRespo
...
+                # Manage a collection creation. The apikey user (or local user if in local mode)
+                # should be the same as the owner field in the body request. In other words, the
+                # apikey user cannot create a collection owned by another user
+                if user != user_login:
+                    detail = {
+                        "error": f"The '{user_login}' user cannot create a \
+collection owned by the '{user}' user",
+                    }
+                    logger.error(detail["error"])
+                    return JSONResponse(content=detail, status_code=HTTP_400_BAD_REQUEST)
```
Comment (Padeanu) [inline services/catalog/rs_server_catalog/user_catalog.py:464]: Status code should be one of 401, 403, **405**.

**Comment by Padeanu** [comment]
Do we have a test where a valid collection is created, then with a PATCH request the owner is changed?

File: services/catalog/rs_server_catalog/user_catalog.py [hunk 2]
```diff
@@ -778,6 +794,16 @@ def manage_delete_request(self, request: Request):
             )
         ):
             return False
+        # we don't care for local mode, any user may create / delete collection owned by another user
+        if common_settings.CLUSTER_MODE and self.request_ids["owner_id"] != user_login:
+            # Manage a collection creation. The apikey user (or local user if in local mode)
```
Comment (jgaucher-cs) [inline services/catalog/rs_server_catalog/user_catalog.py:799]: deletion, not creation

File: services/catalog/rs_server_catalog/user_catalog.py [hunk 3]
```diff
@@ -778,6 +794,16 @@ def manage_delete_request(self, request: Request):
...
+            # should be the same as the owner field in the body request. In other words, the
+            # apikey user can't create a collection owned by another user
```
Comment (jgaucher-cs) [inline services/catalog/rs_server_catalog/user_catalog.py:801]: idem, delete not create

**Comment by agrosu1978** [comment]
> Do we have a test where a valid collection is created, then with a PATCH request the owner is changed?

No, because the PATCH method is not yet implemented

File: services/catalog/rs_server_catalog/user_catalog.py (hunk 2 above)
Comment (agrosu1978) [reply services/catalog/rs_server_catalog/user_catalog.py:799]: Corrected, thanks !

File: services/catalog/rs_server_catalog/user_catalog.py (hunk 3 above)
Comment (agrosu1978) [reply services/catalog/rs_server_catalog/user_catalog.py:801]: Corrected

File: services/catalog/rs_server_catalog/user_catalog.py (hunk 1 above)
Comment (agrosu1978) [reply services/catalog/rs_server_catalog/user_catalog.py:465]: Corrected

**Comment by agrosu1978** [comment]
> Do we have a test where a valid collection is created, then with a PATCH request the owner is changed?

No, because the PATCH method is not yet implemented

**Pull request merged**